Sources/AblyAssetTrackingPublisher/Services/RouteProvider.swift

```diff
@@ -23,8 +23,13 @@ class DefaultRouteProvider: NSObject, RouteProvider {
 
     func changeRoutingProfile(to routingProfile: RoutingProfile, completion: @escaping ResultHandler<Route>) {
         self.routingProfile = routingProfile
-        guard let destination = self.destination,
-              !isCalculating(resultHandler: completion) else {
+        guard let destination = destination else {
+            let errorInformation = ErrorInformation(type: .publisherError(errorMessage: "Cannot change routing profile when destination is not available"))
+            completion(.failure(errorInformation))
```

Comment:
I'm not sure if we should say that the `changeRoutingProfile` operation fails because there is no trackable with a destination active in the publisher at the moment 🤔 You should be able to change the routing profile even if you do not have a single trackable added. I'd say that if there is no destination then you simply change the routing profile and treat it as a success 🤔

Reply:
@KacperKluka In Android example app, this seems to be set only when a publisher is created which is not the case here. I wonder why we do have this option of changing this after creating the publisher on iOS app. In the first page there is already an option of Routing profile which only changes a setting. I'm also not sure about changing the result interface for this - if it's a success it will expect a route which is really not the case here. Is there a spec on what this changing profile button should do @lawrence-forooghian ?

Reply:
> Is there a spec on what this changing profile button should do

As far as I'm aware, there aren't any formal requirements for the example apps, but perhaps @KacperKluka knows otherwise.

> I wonder why we do have this option of changing this after creating the publisher on iOS app.

In my mind, the idea of the publisher example app is to expose the full API of the publisher SDK. The `Publisher.changeRoutingProfile(profile:,completion:)` method doesn’t document any restrictions on when it’s okay to call it, so in the example app I allow the user to call it whenever they want.

Either we need to document restrictions on when this method can be called, or we need to allow it to always be called and make sure the callback is always called.

Reply:
> I'd say that if there is no destination then you simply change the routing profile and treat it as a success 🤔

Is that how it behaves on Android, @KacperKluka? Our aim here should be to be consistent.